app/src/main/java/com/anysoftkeyboard/ime/AnySoftKeyboardWithGestureTyping.java

```diff
@@ -383,6 +447,7 @@ public void onGestureTypingInputDone() {
 
         final GestureTypingDetector currentGestureDetector = mCurrentGestureDetector;
         if (ic != null && currentGestureDetector != null) {
+            Logger.d(TAG, "Completed gesture: %s.", currentGestureDetector.getWorkspaceToString());
```

Comment:
For privacy (and ease-of-use) concerns, I suggest that:
* should be activated by a shared-prefs setting (disabled default).
* should be disabled if in _Incognito_ mode.
* this will be written to a file in the public external storage.

If users will be able to enable this with a click and then easily find the log file that would make it more likely they will contribute.

Reply:
Good point. I meant for it to be used by developers while this pull request is open (with abd logcat), and to remove it in the last commit before merging. I don't know that I'll have time to implement UI changes as I'm not familiar with the codebase at all and Android is not my strong suit anymore so I'll focus on modeling for now and get back to this right before merging.